### PR TITLE
Use PBJ for topics k/v `State` with workflows enabled

### LINKED ISSUE #5111: The key/value types of all `States` must be PBJ objects.

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Updates the key/value state in `ConsensusServiceImpl` schema to use a PBJ `Topic` instead of the mono-service `MerkleTopic`.
    * `Topic` protobuf definition is [here](https://github.com/hashgraph/hedera-protobufs/blob/05111-use-pbj-for-hcs-state/services/state/consensus/topic.proto).
 - Since this invalidates the `MerkleHederaState` `StateChildrenProvider.topic()` implementation (i.e. all `mono-service` use would now result in `ClassCastException`'s), creates a thin adapter for the `GetTopicInfo` query usage estimator.
 - Closes #5111

### CODE REVIEW

**Review by iwsimon — DISMISSED** [review]
LGTM

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/serdes/SerdesFactory.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+
+/**
+ * A factory that creates {@link Serdes} implementations from various ingredients.
+ *
+ * <p>Mostly useful for packaging a PBJ {@code Writer} and {@code ProtoParser} into
+ * a {@link Serdes} implementation.
+ */
+public class SerdesFactory {
```
Comment (mhess-swl) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/serdes/SerdesFactory.java:33]:
```suggestion
public final class SerdesFactory {
```

for a utility class

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/serdes/SerdesFactory.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            public void write(final @NonNull T item, final @NonNull java.io.DataOutput output) throws IOException {
+                if (output instanceof SerializableDataOutputStream out) {
+                    writer.write(item, new DataOutputStream(out));
+                } else {
+                    throw new IllegalArgumentException("Unsupported output type: " + output.getClass());
+                }
+            }
+
+            @Override
+            public int measure(@NonNull java.io.DataInput input) {
```
Comment (mhess-swl) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/serdes/SerdesFactory.java:60]: What does this method do?

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/TemporaryUtils.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * Some temporary utilities to help with the transition from gRPC-generated to
+ * PBJ-generated objects.
+ */
+public class TemporaryUtils {
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/TemporaryUtils.java:39]:
```suggestion
public final class TemporaryUtils {
```

File: settings.gradle.kts [hunk 4]
```diff
@@ -99,7 +99,7 @@ gitRepositories {
     uri.set("https://github.com/hashgraph/hedera-protobufs.git")
     // choose tag or branch of HAPI you would like to test with
     // This version needs to match tha HAPI version below in versionCatalogs
-    tag.set("v0.35.0")
+    tag.set("05111-use-pbj-for-hcs-state")
```
Comment (jeromy-cannon) [inline settings.gradle.kts:102]: update before merge?

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/serdes/SerdesFactory.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/serdes/SerdesFactory.java]: Done!

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/serdes/SerdesFactory.java (hunk 2 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/serdes/SerdesFactory.java:60]: I'm not quite sure yet...just know it's not necessary when dealing with an in-memory key/value type. 😁

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/TemporaryUtils.java (hunk 3 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/TemporaryUtils.java]: Done!

File: settings.gradle.kts (hunk 4 above)
Comment (tinker-michaelj) [reply settings.gradle.kts:102]: Def, before going to `develop` we'll have a long-lived protobufs branch. (Note the target of this PR is another branch, not `develop`.)

**Review by mhess-swl — APPROVED** [review]

**Review by jeromy-cannon — APPROVED** [review]

**Comment by tinker-michaelj** [comment]
Merged via https://github.com/hashgraph/hedera-services/pull/5228

**Pull request closed without merging**